### PR TITLE
Add Microsoft Teams notifier

### PR DESCRIPTION
**Pull request by arneschulz1984**
Some weeks ago, my company decides to use Microsoft Teams.
So we have to migrate from Slack to Microsoft Teams.

After looking around, my preferred solution will be a native Prometheus Alertmanager Notifier.
I do not like Grafana-Alerts so much and an extra container for forwarding the Webhook i also do not like so much.

So pleas have a look at my native implementation of the Microsoft Teams Notifier implementation.

### CODE REVIEW

File: cmd/alertmanager/main.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package main
 import (
 	"context"
 	"fmt"
+	"github.com/prometheus/alertmanager/notify/msteams"
```
Comment (AndreasBergmeier6176) [inline cmd/alertmanager/main.go]: Maybe set up your editor to run `go fmt` on save?

**Comment by simonpasquier** [comment]
Thanks for suggesting this change. Unfortunately, we are currently not accepting new integrations as noted in the [documentation](https://prometheus.io/docs/alerting/configuration/#receiver).

We can only provide the stability and performance we want to provide if we can properly maintain the codebase. This includes, amongst others, test integrations in an automated and scalable fashion. For this reason, we are suggesting people integrate with the help of our generic interfaces. We have an integrations page on which integrations using our generic interfaces are listed.

Even if existing integrations can not be tested in an automated fashion, we will not remove them for reasons of compatibility. This also means that any additions we take on, or any changes to existing integrations we make or accept, will mean maintaining and testing these until at least the next major version, realistically even beyond that.

Feel free to question this answer on our [developer mailing list](https://groups.google.com/forum/#!forum/prometheus-developers), but be aware it's unlikely that you will get a different answer.

**Comment by simonpasquier** [comment]
Closing it given the current moratorium on receivers. We're aware that it isn't a satisfying answer but the reasons listed above still stand for now.

**Comment by mmethner** [comment]
Too bad that the PR was rejected, thanks anyway for the work.

**Pull request closed without merging**